### PR TITLE
feat(labs): support list of notes for lab requests

### LINKED ISSUE #2297: Support a list of notes for a lab request
## 🚀 Feature Proposal
Currently, the lab requests page only supports one note. If there are multiple notes the one note field would have to be constantly updated. 

This feature is to support a list of notes for a given lab request. This should work similar to a comments section does. 

We can display past notes with the callout component. At the bottom of the note list, a text area should appear for adding a new notes.

### PR DESCRIPTION
**Pull request by rsousaj**
Fixes #2297 .

**Changes proposed in this pull request:**

- Lab's interface notes property as string[]
- Callout components used to show past notes at ViewLab

### CODE REVIEW

File: src/labs/ViewLab.tsx [hunk 1]
```diff
@@ -73,7 +74,16 @@ const ViewLab = () => {
       )
     }
     if (labToView) {
-      dispatch(updateLab(labToView, onSuccess))
+      const newLab = labToView as Lab
+      if (newNotes) {
+        newLab.notes = newLab.notes
```
[the remark this answers is not in the document]
Comment (rsousaj) [reply src/labs/ViewLab.tsx:79]: I appreciate your help. Should the notes field on Lab interface be optional ? Cause if it is, TypeScript was complaining about using spread on it.

[the remark this answers is not in the document]
Comment (rsousaj) [reply src/labs/ViewLab.tsx]: Anyway, I've tried a different approach on the newLabRequest that helps to clean these messy code. Could you review it again ? What do you think ?

File: src/labs/requests/NewLabRequest.tsx [hunk 2]
```diff
@@ -24,10 +24,12 @@ const NewLabRequest = () => {
   const [newLabRequest, setNewLabRequest] = useState({
     patient: '',
     type: '',
-    notes: '',
+    // notes: [''],
```
Comment (fox1t) [inline src/labs/requests/NewLabRequest.tsx:27]: Just remove this line

**Review by fox1t — CHANGES_REQUESTED** [review]

File: src/labs/requests/NewLabRequest.tsx (hunk 2 above)
Comment (rsousaj) [reply src/labs/requests/NewLabRequest.tsx]: My bad.

Comment (rsousaj) [reply src/labs/requests/NewLabRequest.tsx]: I did it as requested but looks like the deployment failed and I don't really know if it was my fault.

**Review by fox1t — APPROVED** [review]
LGTM

**Review by blestab — APPROVED** [review]

**Comment by morrme** [comment]
This seems to be affected by https://github.com/HospitalRun/components/issues/583 @rsousaj @jackcmeyer

**Comment by morrme** [comment]
@tehKapa What is the status of this PR?

**Review by jackcmeyer — APPROVED** [review]

**Pull request merged**